### ISSUE TITLE
computegravity.fx doesn't work after downsampling 'hotsampling'

### ISSUE THREAD

**Issue by Defaltsiuncula** [LordKobra/CobraFX#2]
Hey,

Sorry to open this issue, because I don't know if I can call this an issue, but I'm using the computegravity.fx from you in The Witcher 3 Wild Hunt v1.32 game. Everything works correctly before 'hotsampling'. When I tick 'useimage' option it doesn't apply that effect only after 'hotsampling'. So it works before I 'hotsample'. What I could possible do in order to see the effect 'useimage' after 'hotsampling'? Maybe its my ReShade version. I'm using the 5.01, but its mentioned that I need above 4.8.

Much Love

**Comment by LordKobra** [comment]
Hello,

that's a good question! From my personal experience with hotsampling, it can take some time before the ReShade effects are applied. Have you checked if other effects are visible in the screenshot?

**Comment by Defaltsiuncula** [comment]
Thanks for the fast response! Yeah, all other effects are there on the shot after 'hotsampling'. Computegravity.fx effect works as well but only when I don't turn on the 'useimage' option. I'm downloading the latest ReShade to try it out right now. Let me know if you have some idea what could possible be. You kinda want the useimage of course for the shot, because it can stand out more artistically.

**Comment by LordKobra** [comment]
Sadly, i have no idea right now. I guess you are using SRWE for hotsampling? 

I just tried SRWE with ReShade 5.6 in a Testbox and it seems to work with useimage. In a few weeks i got more time, so I could backup the current Witcher installation and revert to the 1.32 beta branch for testing and take a look at the code too.

**Comment by Defaltsiuncula** [comment]
You are correct! I just upgraded my ReShade, and I used only computegravity.fx for 'hotsampling' and it worked! Effect showed while 'useimage' was ticked. I will test more today with other effects and stuff, but it seems that my ReShade version was bad. I was using some modified ReShade version 5.01 so that could be the case of shader not working properly after 'hotsampling'. I mean thats just my guess because it seems that its working properly now.

**Comment by LordKobra** [comment]
That's great to hear! Yes, sometimes older versions have a few issues, especially when a lot of changes got introduced (5.0).

Can the issue be closed or do you have any other problems?

PS: I think Otis_Inf tools have a variant of SRWE included, as he is one of the SRWE project contributors.

**Comment by Defaltsiuncula** [comment]
I would really like to close this issue hahaha, but yeah unfortunately I will need to take time testing this, because it seems on aspect ratio 21:9 I have issue with it (effect not showing again). On 3:4 and resolution of 3750x5000 it shows after couple of seconds literally, but on 21:9 (6840x2930 it never shows, even after 5 mins. On lower resolution for example 6496x2815 it shows instantly, but above it won't. So thats what I can say after 5 mins of testing. 

Yeah SRWE is from the same person. But im using the The Witcher 3 Wild Hunt tools (classic edittion) since SRWE doesn't work with The Witcher 3. I mean I never heared someone used SRWE for The Witcher 3.

**Comment by Defaltsiuncula** [comment]
Okay, thanks for the response its appreciated really. I will play around with the settings, and will let you know my experience later. But whats the positive thing is that it works. Before it didn't at all so yeah I'm really happy.

In some games I can't 'hotsample' that high, but in The Witcher 3 I can, and its kinda a bit odd that I can't 'hotsample' high with this effect. I will try play with the GRAVITY_HEIGHT and see how that will work. Thanks for your explanation. Will let you know. :)

**Comment by Defaltsiuncula** [comment]
Alright I'm back. I did quite some testing and it seems that computegravity.fx loads in high resolution with 768 height, but only if  I deactivate other shaders. When I turn on DPX, AdaptiveFog, Clarity and few others along with gravity and then use 'hotsampling', there is a big chance that gravity won't load. Idk why really. I think it would be the best for you to test it out. I will edit this comment if I manage to get it work. Idk what I'm doing wrong really.

**Comment by Defaltsiuncula** [comment]
When I said how shader ''wont show up'', yeah I meant on the effect after hotsampling. Soon as I exit the hotsampling gravity effect is there active along with other shaders. He does work in hotsampling, but only if he is solo. He works with some DoF shader, but 90% of others won't work with computegravity.fx. And when it come's to errors, I can see some error, but its not gravity.fx. SuperDepth3D_VR+.fx error and Depth_Cues.fx.

**Comment by Defaltsiuncula** [comment]
I tried with other game as well, Remnant From The Ashes in this case, and it works flawlessly with all the shaders on resolution 6840x2930.

**Comment by LordKobra** [comment]
Thanks, very interesting! It *could be* ReShade terminating the compilation. Since it's a compute shader, the compilation can take some time. This depends on many factors. But if the shader takes too long to compile, ReShade can skip it and it won't show up.

**Comment by Defaltsiuncula** [comment]
Update! Since I was using the ReShade 5.5.2 with addons in Remnant, I was like lemme downgrade from 5.6.0 in The Witcher 3 and try that version. And guess what. IT works!!! It compiled instantly with any shader I add. So it was just the ReShade version. It didn't worked with 5.01 and with latest one 5.6.0, but it works with 5.5.2. I mean I kinda knew its not the shader problem, so I was like lemme switch the version. Oh man, amazing !

**Comment by Defaltsiuncula** [comment]
Oh no , I forgot to use the image... Soon as I click in use the image then it doesnt work. Sorry, sometimes I just rush without thinking.

**Comment by LordKobra** [comment]
Ye, don't worry! It will take some time, but i have some ideas now where to look. And I'm quite sure there is one bug in need of fixing anyways.

**Comment by Defaltsiuncula** [comment]
Oki, thats great to hear! You created amazing stuff really, like its so fun to use. 

So in both cases it worked without using the image. When I tick 'useimage' and hotsample then it won't work, even when he is solo active. So 'useimage' is problematic when hotsampling. Thats it for todays testings I will play without use image for now, and I hope you can found the problem in the future.  If there is a problem, maybe my GPU is just weak so yeah.

For me 'useimage' works only if the HEIGHT is 256. 512 and 786 doesn't work for me even if he is solo.

**Comment by Defaltsiuncula** [comment]
Sorry I forgot to close this, since it was never really an issue. Thanks again for the answers and your work! Much Love.

**Comment by Defaltsiuncula** [comment]
Sorry I forgot to close this thread since it was never really an issue. Thanks again for your answers and your work! Much Love

**Comment by LordKobra** [comment]
> Oki, thats great to hear! You created amazing stuff really, like its so fun to use. All CobraFX are just stunning really.
> 
> So in both cases it worked without using the image. When I tick 'useimage' and hotsample then it won't work, even when he is solo active. So 'useimage' is problematic when hotsampling. Thats it for todays testings I will play without use image for now, and I hope you can found the problem in the future. If there is a problem at all, maybe my GPU is just weak so yeah.
> 
> For me 'useimage' works only if the HEIGHT is 256. 512 and 786 doesn't work for me even if he is solo active.

Hello, Gravity_CS.fx has recently been updated to run better and faster at all resolutions and features a hotsampling option. I hope this resolved the issue with "Use Image" you had.

**Comment by Defaltsiuncula** [comment]
> Hello, Gravity_CS.fx has recently been updated to run better and faster at all resolutions and features a hotsampling option. I hope this resolved the issue with "Use Image" you had.

POG! I thought my PC is weak in the end so it cannot handle that really haha so I closed the issue. And like I mentioned idk if it was the issue really, but you said its optimized more now so thats great to hear. Anyways, thanks for letting me know man, you are so kind! I will download today, but will check fx in the future because right now I'm not into taking shots lately. I will though soon because I will have lots of time in like two weeks or so. Thanks again for your work and your time Cobra, I really really appreciate it. Your work is top notch! I've been using your fxs really a lot, especially RealLongExposure. <3

**Comment by Defaltsiuncula** [comment]
Hey, an update to my previous post. I forgot to say but, it was a week ago when I tested gravity in Assetto Corsa, the only game that I kinda play/shot these days in free time, and it worked without issues! I hotsampled to 5760x2468 with 768 height, effect rendered with all the other effects really fast. I haven't tested in other games or in the The Witcher 3 Wild Hunt where I had issues at first, but I assume it will work everywhere now anyways. Will edit my comment to add more of the info if needed. Thanks again!
